### PR TITLE
Add registry-convoy template

### PR DESCRIPTION
**Pull request by sshipway**
This is the same as the Registry template, however it uses Convoy Storage Pool volumes instead of a shared filesystem mountpoint for people who prefer to use Convoy than to have NFS common mounts in their Environment.

### CODE REVIEW

**Comment by cloudnautique** [comment]
@sshipway

Not sure whats going on, but it looks like there are 6 commits instead of just the 1 for the registry. Does it need a rebase?

**Comment by sshipway** [comment]
I'm having problems with this git thing.  I'll have another go at rebasing it, and rebuild the whole fork if I have to.

**Comment by sshipway** [comment]
Success.  Please merge if you're happy with the configuration.

**Comment by jmreicha** [comment]
Looking forward to trying this.  Is there anything holing this up from being merged?

**Comment by sshipway** [comment]
@cloudnautique Successfuly rebased pull request for registry-convoy template addition, hopefully all is OK to merge now.

**Comment by cloudnautique** [comment]
Looks good.

File: templates/registry-convoy/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  image: rancher/load-balancer-service
+  tty: true
+  stdin_open: true
+  ports:
+  - ${RPORT}:5000/tcp
+  - ${PPORT}:443/tcp
+  labels:
+    io.rancher.loadbalancer.target.sslproxy: ${PPORT}=443
+    io.rancher.loadbalancer.target.registry: ${RPORT}=5000
+    io.rancher.scheduler.global: 'true'
```
Comment (jmatsushita) [inline templates/registry-convoy/0/docker-compose.yml]: Hi there, could you explain if there's a reason why the load balancer is scheduled to go on all hosts?

Comment (sshipway) [inline templates/registry-convoy/0/docker-compose.yml]: This is for installations with all Hosts behind a hardware load balancer or common DNS record.  The additional affinity rules following allow you to restrict it to not appear on hosts with lb=0 or registry.enabled=false

**Pull request merged**